pkgs/test_api/lib/src/frontend/throws_matcher.dart

```diff
@@ -33,6 +33,30 @@ const Matcher throws = Throws();
 /// In all three cases, when an exception is thrown, this will test that the
 /// exception object matches [matcher]. If [matcher] is not an instance of
 /// [Matcher], it will implicitly be treated as `equals(matcher)`.
+///
+/// Examples:
+/// ```dart
+/// void functionThatThrows() => throw UnsupportedError('Unsupported');
```

Comment:
[nit] Testing behavior of `*Exception` being thrown should hopefully be more common than testing that a `*Error` is thrown. Can we maybe use some subtype of exception, or just a fake `SomeException` type?

Reply:
Fair point.  Done.